### DISCUSSION TITLE
Get Azure SSO "auto create user" to use attribute sAMAccountName instead of name for user name?

### DISCUSSION THREAD

**Discussion by paulrosberg** [netbox-community/netbox#15329] [Q&A]
So I set up Azure SSO using https://docs.netbox.dev/en/stable/administration/authentication/microsoft-azure-ad/ and works great. The only thing I want to change is the fact the the auto created users are using the name attribute instead of sAMAccountName (users are coming from our Active Directory).
ex.
If name attribute is "John Dow" the auto created username is "JohnDow".
Is it possible to get NetBox to instead use the sAMAccountName for the user name? Was trying to find a way to change the attribute mapping in Azure for NetBox but failed to do so. Anyone know?

**Comment by miked314** [comment]
Did you happen to solve this issue?

**Comment by ghost** [comment]
Also interested in this answer.

**Reply by miked314** [reply]
@goteamkor 

if you are using social_core.backends.azuread solution is to setup custom pipeline. 


in the same folder as configuration.py file create custom_pipeline.py (replace yourcompanydomainname)
```
def set_username(response, *args, **kwargs):
    return { 'username': response['upn'].lower().replace("@yourcompanydomainname","") }
```


configuration.py:

```
SOCIAL_AUTH_PIPELINE = (
    'social_core.pipeline.social_auth.social_details',
    'social_core.pipeline.social_auth.social_uid',
    'social_core.pipeline.social_auth.auth_allowed',
    'social_core.pipeline.social_auth.social_user',
    'social_core.pipeline.user.get_username',
    # Look up existing users by E-mail. As we trust our Corp Azure AD
    'social_core.pipeline.social_auth.associate_by_email',
    'social_core.pipeline.user.create_user',
    'social_core.pipeline.social_auth.associate_user',
    # Your custom pipeline below
    'netbox.custom_pipeline.set_username',
    # Enable loading of extra scopes (e.g. roles)
    'social_core.pipeline.social_auth.load_extra_data',
    'social_core.pipeline.user.user_details',
    # Debug the pipeline - Only do this for a test-instance!!!
    # 'social_core.pipeline.debug.debug'
)
```
